protostar/commands/upgrade/upgrade_command.py

```diff
@@ -89,9 +88,10 @@ def upgrade(self):
             self.current_version,
             self.latest_version,
         )
+
+        self._pull_tarball()
```

Comment:
Why did you move this call out of `try`'s scope?

Reply:
Because i have to pull before backup, and there is nothing to cleanup if it fails.

Reply:
backup moves cert location

Reply:
good point 😅